File: leetcode/js/maximum-sum-with-exactly-k-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximizeSum = function (nums, k) {
  let result = 0;
  const max = Math.max(...nums);

  for (let i = 0; i < k; i++) {
    result += max + i;
  }

  return result;
};

var maximizeSum = function (nums, k) {
  return k * (Math.max(...nums) + (k - 1) / 2);
};

File: leetcode/rust/maximum-sum-with-exactly-k-elements.rs
impl Solution {
    pub fn maximize_sum(nums: Vec<i32>, k: i32) -> i32 {
        let mut result = 0;
        let max_num = nums.iter().max().unwrap();

        for i in 0..k {
            result += max_num + i;
        }

        result
    }
}

impl Solution2 {
    pub fn maximize_sum(nums: Vec<i32>, k: i32) -> i32 {
        if let Some(&max) = nums.iter().max() {
            return max * k + k * (k - 1) / 2;
        } else {
            return 0;
        }
    }
}
